packages/snaps-rpc-methods/src/restricted/dialog.ts

```diff
@@ -230,17 +298,23 @@ function getValidatedParams(
     return create(params, struct);
   } catch (error) {
     if (error instanceof StructError) {
-      const { key, type: errorType } = error;
+      const { failures } = error;
+
+      const failureArray = failures();
 
-      if (key === 'placeholder' && errorType === 'never') {
+      const placeholderFailure = failureArray.find(
+        ({ key, type: errorType }) =>
+          key === 'placeholder' && errorType === 'never',
+      );
+      if (placeholderFailure) {
         throw rpcErrors.invalidParams({
           message:
             'Invalid params: Alerts or confirmations may not specify a "placeholder" field.',
         });
       }
 
       throw rpcErrors.invalidParams({
-        message: `Invalid params: ${error.message}.`,
+        message: `Invalid params: ${getFirstErrorInUnion(failureArray)}.`,
```

Comment:
This is interesting. I want to get @Mrtenz opinion on whether there is a better solution for improving union error messages across the repo.

For context, Maarten, we are seeing an increased amount of error messages that are equivalent to: 
```Expected the value to satisfy a union of `object | object`, but received: [object Object].```

Reply:
Yes but there extract all the error messages and it's pretty ugly, I needed one that just extrats the first error after the union error

![image](https://github.com/MetaMask/snaps/assets/13910212/ae2c44cb-09fe-4d44-a4b7-ddc795b5d2f9)